### PR TITLE
wip: collect inline require statements and convert them to top level imports

### PR DESCRIPTION
**Pull request by n1ru4l**
It might be possible that this should be better off within relay. however, this works great for my use-cases as a quick fix.

Example output:

```diff
/* tslint:disable */
/* eslint-disable */
// @ts-nocheck

+ import tokenInfoSideBar_MoreNotesQuery from "./tokenInfoSideBar_MoreNotesQuery.graphql"
import { ReaderFragment } from "relay-runtime";
import { FragmentRefs } from "relay-runtime";
export type tokenInfoSideBar_NotesFragment = {
  readonly notes: {
    readonly __id: string;
    readonly edges: ReadonlyArray<{
      readonly node: {
        readonly id: string;
        readonly documentId: string;
        readonly title: string;
      };
    }>;
    readonly pageInfo: {
      readonly endCursor: string;
      readonly hasNextPage: boolean;
    };
  };
  readonly " $refType": "tokenInfoSideBar_NotesFragment";
};
export type tokenInfoSideBar_NotesFragment$data = tokenInfoSideBar_NotesFragment;
export type tokenInfoSideBar_NotesFragment$key = {
  readonly " $data"?: tokenInfoSideBar_NotesFragment$data;
  readonly " $fragmentRefs": FragmentRefs<"tokenInfoSideBar_NotesFragment">;
};

const node: ReaderFragment = (function () {
  var v0 = ["notes"];
  return {
    argumentDefinitions: [
      {
        defaultValue: 20,
        kind: "LocalArgument",
        name: "count",
      },
      {
        defaultValue: null,
        kind: "LocalArgument",
        name: "cursor",
      },
      {
        defaultValue: null,
        kind: "LocalArgument",
        name: "filter",
      },
      {
        defaultValue: null,
        kind: "LocalArgument",
        name: "key",
      },
    ],
    kind: "Fragment",
    metadata: {
      connection: [
        {
          count: "count",
          cursor: "cursor",
          direction: "forward",
          path: v0 /*: any*/,
        },
      ],
      refetch: {
        connection: {
          forward: {
            count: "count",
            cursor: "cursor",
          },
          backward: null,
          path: v0 /*: any*/,
        },
        fragmentPathInResult: [],
-        operation: require("./tokenInfoSideBar_MoreNotesQuery.graphql.ts"),
+        operation: tokenInfoSideBar_MoreNotesQuery,
      },
    },
    name: "tokenInfoSideBar_NotesFragment",
    selections: [
      {
        alias: "notes",
        args: [
          {
            kind: "Variable",
            name: "__dynamicKey",
            variableName: "key",
          },
        ],
        concreteType: "NoteConnection",
        kind: "LinkedField",
        name: "__tokenInfoSideBar_notes_connection",
        plural: false,
        selections: [
          {
            alias: null,
            args: null,
            concreteType: "NoteEdge",
            kind: "LinkedField",
            name: "edges",
            plural: true,
            selections: [
              {
                alias: null,
                args: null,
                concreteType: "Note",
                kind: "LinkedField",
                name: "node",
                plural: false,
                selections: [
                  {
                    alias: null,
                    args: null,
                    kind: "ScalarField",
                    name: "id",
                    storageKey: null,
                  },
                  {
                    alias: null,
                    args: null,
                    kind: "ScalarField",
                    name: "documentId",
                    storageKey: null,
                  },
                  {
                    alias: null,
                    args: null,
                    kind: "ScalarField",
                    name: "title",
                    storageKey: null,
                  },
                  {
                    alias: null,
                    args: null,
                    kind: "ScalarField",
                    name: "__typename",
                    storageKey: null,
                  },
                ],
                storageKey: null,
              },
              {
                alias: null,
                args: null,
                kind: "ScalarField",
                name: "cursor",
                storageKey: null,
              },
            ],
            storageKey: null,
          },
          {
            alias: null,
            args: null,
            concreteType: "PageInfo",
            kind: "LinkedField",
            name: "pageInfo",
            plural: false,
            selections: [
              {
                alias: null,
                args: null,
                kind: "ScalarField",
                name: "endCursor",
                storageKey: null,
              },
              {
                alias: null,
                args: null,
                kind: "ScalarField",
                name: "hasNextPage",
                storageKey: null,
              },
            ],
            storageKey: null,
          },
          {
            kind: "ClientExtension",
            selections: [
              {
                alias: null,
                args: null,
                kind: "ScalarField",
                name: "__id",
                storageKey: null,
              },
            ],
          },
        ],
        storageKey: null,
      },
    ],
    type: "Query",
    abstractKey: null,
  };
})();
(node as any).hash = "5e7babc5b768b840540fee0d178b54cd";
export default node;

```

### CODE REVIEW

**Comment by n1ru4l** [comment]
@sibelius It seems actions don't run for pushes from fork repos

**Comment by sibelius** [comment]
how can we enable this?
can we remove everything from travis?

I've gave you access to this org and repo

**Comment by n1ru4l** [comment]
> can we remove everything from travis?

I think publishing is automated  via travis cc @alloy , we could move it to github actions.

> how can we enable this?

I will check later

**Comment by alloy** [comment]
Agreed that this should normally go into relay, but I’m not against doing it here temporarily 👍 

I assume doing it with regexes is faster than using a babel visitor on the JSON, but it might be nice to compare, if you feel like it (don't feel strong on this at all).

**Review by sibelius — APPROVED** [review]
as this is the same as flow compiler, we can follow this approach

**Comment by n1ru4l** [comment]
@sibelius The flow compiler does not address this issue. In case you are referring to https://github.com/facebook/relay/pull/2841/files#diff-8258e3677c0534bd73d878b16093be514b2227f8972766d881bf903ff5f0e0d9R15-R27 - that was only a WIP pr that got superseded by https://github.com/facebook/relay/pull/2781 which never addressed this issue (maybe because the `@refetchable` problematic has been introduced at a later point).

**Comment by rubas** [comment]
- little bump - 

Anything that is needed here @n1ru4l to fix this for the moment?

(We use snowpack/webpack pipeline and the current require command even hardcodes the whole filename (foo.graphql.**ts**) which breaks our pipeline.)

**Comment by n1ru4l** [comment]
@rubas I want to merge this together with the major release: https://github.com/relay-tools/relay-compiler-language-typescript/pull/275

I have limited time right now and this is not high in my priority list.

**Comment by theseyi** [comment]
> @rubas I want to merge this together with the major release: #275

Since #275 got merged, will there be any progress to merge this as well? @n1ru4l thanks

**Comment by n1ru4l** [comment]
@theseyi You wanna write some tests for this?

File: src/formatGeneratedModule.ts [hunk 1]
```diff
@@ -2,6 +2,36 @@ import { FormatModule } from "relay-compiler";
 import * as ts from "typescript";
 import addAnyTypeCast from "./addAnyTypeCast";
 
+const createRequireRegex = () => /require\("(.*)"\)/g;
```
Comment (asterikx) [inline src/formatGeneratedModule.ts:5]: @n1ru4l The regex used to replace require calls is off. In my case, the paths of the required modules are delimited by single quotes, whereas the regex looks for double quotes.

E.g.: The regex will not identify the require call in the following line: `"operation": require('./BlogPaginationQuery.graphql.ts')`

After replacing the double quotes in the regex by single quotes, require calls were successfully replaced for me.

Comment (n1ru4l) [reply src/formatGeneratedModule.ts:5]: @asterikx you could help pushing this getting merged by creating a pr that adds tests to this branch

Comment (asterikx) [reply src/formatGeneratedModule.ts:5]: @n1ru4l Can you have a look at https://github.com/n1ru4l/relay-compiler-language-typescript/pull/1? If I can help any further to get this merged, let me know.

Comment (n1ru4l) [reply src/formatGeneratedModule.ts:5]: @asterikx I did not see the PR, awesome! I will check it out next week!

**Comment by n1ru4l** [comment]
@asterikx Do you mind-altering the README to include a section on how to properly configure typescript and relay config in order to produce esm imports? After that is done I will merge this! Thank you so much for adding the tests

**Pull request merged**